docs/proposals/multi-platform.md

```diff
@@ -0,0 +1,493 @@
+# Kubernetes for multiple platforms
+
+**Author**: Lucas Käldström ([@luxas](https://github.com/luxas))
+
+**Status** (8th of August 2016): Some parts are already implemented; but still there quite a lot of work to be done.
+
+## Abstract
+
+Kubernetes is written in Go, and Go code is portable across platforms.
+Docker and rkt are also written in Go, and it's already possible to use docker on various platforms.
+When it's possible to run docker containers on a specific architecture, people also want to use Kubernetes to manage the containers.
+
+We obviously want Kubernetes to run on as many platforms as possible.
+This is a proposal that explains what already has been done and why, and what's proposed to yet be done.
+
+With this work, we want to achieve a true cross-platform container management system.
+
+In this proposal, `platform` is defined as `operating system/architecture`
+
+The following platforms are proposed to be built for a Kubernetes release:
+ - linux/amd64
+ - linux/arm
+ - linux/arm64
+ - linux/ppc64le
+
+If there's interest in running Kubernetes on `linux/s390x` too, it won't require many changes to the source now when we've laid the ground for a multi-platform Kubernetes already.
+
+There is also work going on with porting Kubernetes to Windows (`windows/amd64`). See [this issue](https://github.com/kubernetes/kubernetes/issues/22623) for more details.
+
+But note that when porting to a new OS like windows, a lot of os-specific changes have to be implemented before cross-compiling, releasing and other concerns this document is describing may apply.
+
+## Motivation
+
+TODO
+
+## Use Cases
+
+TODO
```

Comment:
There're the use cases from China that: both PowerLinux & X64 will be deployed in the data center, it's better for k8s to have the ability to manage those hardware in one cluster.

There're also some other requirements on how to manage heterogeneous environment, but this's the first step :).